homeassistant/components/madvr/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""The madvr-envy integration."""
+
+from __future__ import annotations
+
+import logging
+
+from madvr.madvr import Madvr
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+
+from .coordinator import MadVRCoordinator
+
+PLATFORMS: list[Platform] = [Platform.REMOTE]
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: MadVRConfigEntry) -> bool:
+    """Set up the integration from a config entry."""
+    madVRClient = Madvr(
+        host=entry.data[CONF_HOST],
+        logger=_LOGGER,
+        port=entry.data[CONF_PORT],
+        connect_timeout=10,
+        loop=hass.loop,
+    )
+    coordinator = MadVRCoordinator(hass, madVRClient)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.runtime_data = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    entry.add_update_listener(async_reload_entry)
+
+    # handle loading operations
+    await coordinator.handle_coordinator_load()
+    return True
+
+
+async def async_handle_unload(coordinator: MadVRCoordinator) -> None:
```

Comment:
Let's also listen to HOME_ASSISTANT_STOP on the event bus and execute this function